federation/pkg/kubefed/init/init.go

```diff
@@ -582,6 +601,42 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 	return clientset.Extensions().Deployments(namespace).Create(dep)
 }
 
+func waitForPods(clientset *client.Clientset, name, nameSpace string) error {
+	err := wait.PollInfinite(podWaitInterval, func() (bool, error) {
+		podList, err := clientset.Core().Pods(nameSpace).List(api.ListOptions{})
+		if err != nil {
+			return false, nil
+		}
+		for _, pod := range podList.Items {
+			if strings.Contains(pod.Name, name) && pod.Status.Phase == "Running" {
```

Comment:
will update to use HasPrefix now and update rest of the checking method in a separate PR.